data/cookbook/parse-command-line-arguments/00-stdlib.ml

```diff
@@ -0,0 +1,65 @@
+---
+packages: []
+---
+(* `Sys.argv` is the command-line arguments as an array of strings
+   ([standard library documentation](https://ocaml.org/manual/5.3/api/Sys.html#VALargv)).
+   Here we print each argument with its corresponding index in the array. *)
+let simplest_parser () =
+  Sys.argv
+  |> Array.to_list
+  |> List.iteri (Printf.printf "argument %d: %s\n")
+
+
+(* `Arg` is a module for parsing command-line arguments, and it is part of
+   OCaml's standard library
+   ([documentation](https://ocaml.org/manual/5.3/api/Arg.html)).
+   In this function we define the structure of the command-line arguments we
+   expect, the argument types types and their documentation. This is basically
+   the same function defined in the module's documentation.
+
+ *)
+let arg_module_parser () =
+  let usage_msg =
+    "mycmd [--verbose] <file1> [<file2>] ... -o <output>"
+  and verbose = ref false
+  and input_files = ref []
+  and output_file = ref "" in
+  (* This function is called once for each anonymous argument. *)
```

Comment:
`ood-gen` is parsing this whole block as a `Str.Text`, except for the final delimiter `*)`. There seems to be something wrong with the string splitting ([`Str.full_split`, line 72](https://github.com/boechat107/ocaml.org/blob/cookbook-parse-command-line/tool/ood-gen/lib/cookbook.ml#L72)).

Reply:
Ok, got it. The regex expect the delimiter `(*` always the beginning of the line. This code has indented comments.

Reply:
I couldn't find any comments about this comments restriction. Would it make sense to add a comment? Or would it be better to change the regex pattern?